### ISSUE TITLE
Safari - extra transceiver added by createOffer() if we already have a "recvonly" one

### ISSUE THREAD

**Issue by marcin-bazyl** [webrtcHacks/adapter#945]
- [x] I have provided steps to reproduce (e.g. a link to a [jsfiddle](https://jsfiddle.net/))
- [x] I have provided browser name, version and adapter.js version
- [x] This issue only happens when adapter.js is used

## Versions affected

**Browser name including version**
Safari 12.1

**adapter.js**
7.2.2

## Description
The function **shimCreateOfferLegacy()** in **safari_shim.js** is adding an audio transceiver if offerOptions.offerToReceiveAudio is true and it fails to find transceiver with a sender set and kind='audio'. This seems wrong as the user may have a "recvonly" audio transceiver and so there is no need to add another transceiver. The same applies to video transceivers.

The issue only affects Safari as the code is in safari_shim.js

## Steps to reproduce
It's easy to reproduce:
1. Create an RTCPeerConnection 
2. Add 1 audio transceiver with direction "recvonly"
3. Call createOffer() with offerToReceiveAudio=true 

observe that you end up with 2 audio transceivers

link to JSFiddle with example code: https://jsfiddle.net/xerk5sv7/1/

## Expected results
There should be just 1 transceiver after the call to createOffer() in this case.

## Actual results
There are 2 audio transceivers if you run the code in Safari, 1 if using other browsers (for example Chrome)

**Comment by fippo** [comment]
I don't think its a good idea to mix addTransceiver with the legacy offerOptions but...

```
      const audioTransceiver = this.getTransceivers().find(transceiver =>
        transceiver.sender.track &&
        transceiver.sender.track.kind === 'audio');
```
This should probably look at the receiver instead of the sender, as the receiver will always have a track and a kind.

@youennf thoughts?

**Comment by fippo** [comment]
Tentative fix in #946 -- probably needs to check if getTransceivers() always returns an object with sender+receiver even in the weird "plan-b transceiver" version

**Comment by fippo** [comment]
This should be fixed (without dropping http support) in v7.2.3 which just landed on npm as well as adapter-latest.js

Please holler if wrong
